### PR TITLE
fix #11616 Markpoint's information on stack bar is incorrect

### LINKED ISSUE #11616: Markpoint's information on stack bar is incorrect
### Version
4.5.0-rc.2

### Steps to reproduce
Open this gallery https://gallery.echartsjs.com/editor.html?c=xUmiTvvyv&v=1
Max and min for 'cheese cocoa ' are wrong

### What is expected?
'Cheese cocoa' 's max is 124.1, min is 67.2

### What is actually happening?
max = 67.2, min = 86.4

### PR DESCRIPTION
**Pull request by susiwen8**
Add two new methods for get storage data(maybe this kind of methods has existed)

Close #11616

### CODE REVIEW

File: src/component/marker/markerHelper.js [hunk 1]
```diff
@@ -68,8 +68,19 @@ function markerTypeCalculatorWithExtent(
     var value = numCalculate(data, calcDataDim, mlType);
 
     var dataIndex = data.indicesOfNearest(calcDataDim, value)[0];
-    coordArr[otherCoordIndex] = data.get(otherDataDim, dataIndex);
-    coordArr[targetCoordIndex] = data.get(targetDataDim, dataIndex);
+    var targetData = data.getStorageItem(targetDataDim);
+    if (mlType === 'max') {
+        coordArr[targetCoordIndex] = Math.max(...targetData);
```
Comment (pissang) [inline src/component/marker/markerHelper.js]: ES6 syntax here.  Check https://echarts.apache.org/en/coding-standard.html#language-features

File: src/data/List.js [hunk 2]
```diff
@@ -467,6 +467,21 @@ listProto.getProvider = function () {
     return this._rawData;
 };
 
+// Get store data
+listProto.getStorage = function () {
```
Comment (pissang) [inline src/data/List.js]: I don't see `getStorage` is used. And I think it's a good idea to expose `storage` to other modules. All data management should be in the `List` module.

File: src/data/List.js [hunk 3]
```diff
@@ -467,6 +467,21 @@ listProto.getProvider = function () {
...
+};
+
+/**
+ *  Get specific item's data
+ *
+ * @param {string} dim dim name
+ * @return {Array} item data
+ */
+listProto.getStorageItem = function (dim) {
+    return [].slice.call(this._storage[dim][0]);
```
Comment (pissang) [inline src/data/List.js]: `this._storage[dim]` will be separated into sevaral chunks when data is dynamically appended with `appendData` method.
 
Should not hard code the `chunkIndex` to `0` here.

**Comment by pissang** [comment]
Hi, I've leaved comments on the code.

Also the `min` `max` calculation is already in the https://github.com/apache/incubator-echarts/pull/11617/files#diff-ed976f176ba3ed927f196407ddc456a0L68 . I think the fix should be in the function `numCalculate`. And there is no need to copy all the data out of the `List` module.

**Comment by susiwen8** [comment]
@pissang I think `numCalculate` has no logical problem, only the parameter. When the chart is stacked, it returns max or min of the stacked data, and then using `indicesOfNearest` to get data index. So just set dimension to target dimension will fix this.

**Pull request closed without merging**